src/checker.rs

```diff
@@ -293,6 +306,68 @@ pub trait Checker<M: Model> {
         self.discoveries().remove(name)
     }
 
+    /// Wait for all threads to finish whilst reporting, reporting the finish more accurately than
+    /// the interval used for the reporting.
+    fn join_and_report<R>(mut self, reporter: &mut R) -> Self
```

Comment:
Any reason this shouldn't be simply replace `report`?

Reply:
I don't think so, as long as there weren't some use cases where they needed to be separate?